packages/injected/src/ariaSnapshot.ts

```diff
@@ -505,27 +517,44 @@ function buildByRefMap(root: AriaNode | undefined, map: Map<string, AriaNode> =
   return map;
 }
 
-function hasIframeNodes(root: AriaNode): boolean {
-  if (root.role === 'iframe')
-    return true;
-  return (root.children || []).some(child => typeof child !== 'string' && hasIframeNodes(child));
-}
+function compareSnapshots(ariaSnapshot: AriaSnapshot, previousSnapshot: AriaSnapshot | undefined): Map<AriaNode, 'skip' | 'same' | 'changed'> {
+  const previousByRef = buildByRefMap(previousSnapshot?.root);
+  const result = new Map<AriaNode, 'same' | 'changed'>();
 
-function arePropsEqual(a: AriaNode, b: AriaNode): boolean {
-  const aKeys = Object.keys(a.props);
-  const bKeys = Object.keys(b.props);
-  return aKeys.length === bKeys.length && aKeys.every(k => a.props[k] === b.props[k]);
-}
+  // Returns whether ariaNode is the same as previousNode.
+  const visit = (ariaNode: AriaNode, previousNode: AriaNode | undefined): boolean => {
+    let same: boolean = ariaNode.children.length === previousNode?.children.length && ariaNodesEqual(ariaNode, previousNode);
+    if (ariaNode.role === 'iframe')
+      same = false;
 
-export function renderAriaTree(ariaSnapshot: AriaSnapshot, publicOptions: AriaTreeOptions, previous?: AriaSnapshot): string {
-  if (hasIframeNodes(ariaSnapshot.root))
-    previous = undefined;
+    for (let childIndex = 0 ; childIndex < ariaNode.children.length; childIndex++) {
+      const child = ariaNode.children[childIndex];
+      const previousChild = previousNode?.children[childIndex];
+      if (typeof child === 'string') {
+        same &&= child === previousChild;
+      } else {
+        let previous = typeof previousChild !== 'string' ? previousChild : undefined;
+        if (child.ref)
+          previous = previousByRef.get(child.ref);
```

Comment:
The way I read it, I can reorder child nodes and it'll yield the truthy result, yet you have it tested. Why does it work?